core/sr-io/without_std.rs

```diff
@@ -385,6 +385,20 @@ pub mod ext {
 		/// - nonzero otherwise.
 		fn ext_submit_transaction(data: *const u8, len: u32) -> u32;
 
+		/// Returns information about the local node's network state.
+		///
+		/// # Returns
+		///
+		/// - The encoded `Result<offchain::LocalNetworkState, ()>`.
```

Comment:
That's a good question and I cannot answer it properly because I wasn't the most involved person in designing and adding the externalities.

About the single syscall-like entrypoint: I actually had some thoughts about this kind of design for srml-contracts wasm environment. This kind of build up would add some inflexibility and overhead into the substrate host interface. This doesn't just harms the compilation efficiency, but also introduces more high-level problems. For instance, if your runtime wants to call a function that takes two buffers and you have them at hand you cannot pass them directly and would have to encode them probably (re)allocating more than once.

Also, there is another special thing about this design: it has properties of dynamic dispatch. The default way of interaction between a wasm instance and the host is by resolving **all** imports at the instantiation time and if the instance imports something that doesn't exist the instance cannot be instantiated. On the other hand, with such a dynamic dispatch it would instantiate successfully (cause it imports only the syscall entrypoint) but when it would fail only at the runtime when it tries to call a non-supported function. In general this can be a feature or an issue, but I am inclined to think that in the case of substrate it is more an issue since I doubt that there is a lot to do with conditional logic (the runtime should behave the same regardless the set of available functions).

But yeah, it would definitely make the interface simpler.
As a reference, there is https://github.com/paritytech/substrate/issues/2334 to achieve the same (but arguably more complex than what you propose)